prime-dotnet-webapi/Services/ReportingService.cs

```diff
@@ -0,0 +1,117 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Prime.Models;
+using Prime.HttpClients;
+
+namespace Prime.Services
+{
+    public class ReportingService : BaseService, IReportingService
+    {
+        private const int CalculationPeriodInDays = 14;
+
+        private readonly ICollegeLicenceClient _collegeLicenceClient;
+
+        public ReportingService(
+            ApiDbContext context,
+            ILogger<ReportingService> logger,
+            ICollegeLicenceClient collegeLicenceClient)
+            : base(context, logger)
+        {
+            _collegeLicenceClient = collegeLicenceClient;
+        }
+
+        public async Task<string> PopulatePractitionerTableAsync(DateTime? startDate = null, DateTime? endDate = null)
+        {
+            var result = "";
+            try
+            {
+                if (startDate == null)
+                {
+                    startDate = DateTime.Now.AddDays(-CalculationPeriodInDays);
+                }
+
+                if (endDate == null)
+                {
+                    endDate = DateTime.Now;
+                }
+
+                // get all approved enrollee
+                var enrolleeCerts = _context.Enrollees
```

Comment:
Even though the table is misnamed `Certifications`, we should keep "cert" out of the variable names:  `enrolleeLicences` instead?

Reply:
variable name updated.